sfputil/main.py

```diff
@@ -687,123 +693,247 @@ def eeprom(port, dump_dom, namespace):
 
     click.echo(output)
 
+
 # 'eeprom-hexdump' subcommand
 @show.command()
-@click.option('-p', '--port', metavar='<port_name>', required=True, help="Display SFP EEPROM hexdump for port <port_name>")
+@click.option('-p', '--port', metavar='<port_name>', help="Display SFP EEPROM hexdump for port <port_name>")
 @click.option('-n', '--page', metavar='<page_number>', help="Display SFP EEEPROM hexdump for <page_number_in_hex>")
 def eeprom_hexdump(port, page):
-    """Display EEPROM hexdump of SFP transceiver(s) for a given port name and page number"""
-    output = ""
+    """Display EEPROM hexdump of SFP transceiver(s)"""
+    if port:
+        if page is None:
+            page = 0
+        else:
+            page = validate_eeprom_page(page)
+        return_code, output = eeprom_hexdump_single_port(port, page)
+        click.echo(output)
+        sys.exit(return_code)
+    else:
+        if page is not None:
+            page = validate_eeprom_page(page)
+        logical_port_list = natsorted(platform_sfputil.logical)
+        lines = []
+        for logical_port_name in logical_port_list:
+            return_code, output = eeprom_hexdump_single_port(logical_port_name, page)
+            if return_code != 0:
+                lines.append(f'EEPROM hexdump for port {logical_port_name}')
+                lines.append(f'{EEPROM_DUMP_INDENT}{output}\n')
+                continue
+            lines.append(output)
+        click.echo('\n'.join(lines))
 
-    if platform_sfputil.is_logical_port(port) == 0:
-        click.echo("Error: invalid port {}".format(port))
-        print_all_valid_port_values()
-        sys.exit(ERROR_INVALID_PORT)
 
-    if page is None:
-        page = '0'
+def validate_eeprom_page(page):
```

Comment:
@Junchao-Mellanox can we use click option to validate page?

Reply:
This is for backward compatible. Before the PR, we allowed None here and translate None to 0.

Reply:
@Junchao-Mellanox can't we use https://github.com/sonic-net/sonic-utilities/blob/master/sfputil/main.py#L1660 here? What backward compatibility is broken?

Reply:
Please check this line: https://github.com/sonic-net/sonic-utilities/blob/19ea8493388536921ad204833157dac2cd2bf3ba/sfputil/main.py#L693. Page option is an optional option, so it allows None.

Reply:
@Junchao-Mellanox use something like below:-
`@click.option('-n', default=0, '--page', metavar='<page_number>', help="Display SFP EEEPROM hexdump for <page_number_in_hex>")`

Reply:
@Junchao-Mellanox then you don't need to validate page#